.gitbook/3-protocol-devs/1-dep-specs/1-user-flow.md

```diff
@@ -13,23 +13,68 @@
 
 ## Synopsis
 
-TODO: Blah blah...
+This document specified the entire user flow in the Data Exchange Protocol (hereafter 'DEP').
+This document defines the key players essential to operate the DEP, and describes how the players interact to complete the whole data exchange process.
+
+It is recommended to walk through this document first before diving deeply into detailed specifications.
 
 ### Motivation
 
-TODO: Blah blah...
+DEP aims to establish a reliable communication layer between two parties who do not have strong trust in each other in decentralized environments.
+To achieve this goal without centralized/authorized mediators, DEP introduces four key players: data consumer, provider, oracle, and chain validator. These key players collaborate to verify all aspects of the data exchange to minimize security/privacy vulnerabilities.
 
 ### Definitions
 
-- `Data Provider`: TODO
-- TODO
+#### Key Players
+
+##### Data Consumer
+A data consumer is an individual or an organization who wants to consume certain kinds of data for specific purposes, with or without paying.
+
+##### Data Provider
+A data provider is an individual or an organization that holds data and the permission to provide data to data consumers to obtain benefits, such as incentives or services.
+
+##### Oracle (Oracle operator)
+An oracle is a data validator that guarantees data validity and integrity while data is delivered from data providers to data consumers.
+Data verification is essential to ensure the atomicity of data delivery and incentive payments.
+
+##### Panacea Blockchain Validator
+A Panacea blockchain validator is a blockchain node operator that participates in the blockchain consensus process to guarantee the integrity of the whole process of DEP, which includes consuming data, providing data, validating data, and executing payments.
 
 
 ## Technical Specification
 
-### User Flow
+![](images/user-flow.drawio.png)
+
+The diagram describes each key player interacts with each other to complete a cycle of data exchange. The whole process consists of several steps below.
+
+### Data Deal Registration
+
+A data consumer creates a data deal, specifying data schema, a budget in MED, the maximum number of data to consume, and agreement terms. The data consumer should publish the data deal to a public and secure environment, such as a public blockchain, to ensure that the data deal is not changed by anyone else after it is created. In this document, the Panacea blockchain is used for that public environment. For more details of the data deal, please see the [Data Deal](2-data-deal.md) specification.
```

Comment:
In the far future, can someone create a deal with other cryptocurrency, not MED?
I have imagined 😂

Reply:
I think, yes :)